app/src/main/java/ch/epfl/sweng/hostme/ui/search/SearchFragment.java

```diff
@@ -66,6 +70,18 @@ protected void onBindViewHolder(@NonNull ViewHolder holder, int position, @NonNu
                 holder.price.setText(String.format("%s CHF/month", model.getRent()));
                 holder.area.setText(String.format("%s m²", model.getArea()));
                 retrieveAndDisplayImage(holder, model);
+                holder.itemView.setOnClickListener(view -> {
+                    Intent intent = new Intent(getContext(), DisplayApartment.class);
+                    intent.putExtra(ADDR, model.getAddress());
+                    intent.putExtra(RENT, model.getRent());
+                    intent.putExtra(AREA, model.getArea());
+                    intent.putExtra(LID, model.getLid());
+                    //String lease = DateFormat.format("dd-MM-yyyy", model.getCurrentLease()).toString();
+                    intent.putExtra(LEASE, model.getCurrentLease());

```

Comment:
I see you have some commented code that messes with the string representation of the lease Timestamp. The default `toString()` is indeed somewhat large, but passing the actual Timestamp object displays no text at all because `DisplayApartment.java` is expecting a String extra from the Intent. Feel free to put in a temporary resolution or leave it as is, and we can backlog the date formatting - I just wanted to make sure you were aware of the current behavior displaying empty text (ex. `Current Lease: `) in the Display Apartment view.

Reply:
Yes nothing is displayed in the current Lease, we will fix this when we will merge our work.
Thanks for the review !